core/templates/pages/learner-group-pages/create-group/add-syllabus-items.component.html

```diff
@@ -0,0 +1,206 @@
+          {{ categoryButtonText | truncate:14 }}
+        </button>
+        <ul ngbDropdownMenu
+            class="oppia-search-bar-dropdown-menu oppia-search-bar-dropdown-menu-section dropdown-menu"
+            role="menu">
+          <ng-container *ngFor="let item of selectionDetails.categories.masterList; trackBy: index">
+            <li ngbDropdownItem>
+              <a (click)="toggleSelection('categories', item.id)" *ngIf="!first && !last" class="dropdown-item">
+                <span class="protractor-test-deselected">{{ item.text | translate }}</span>
+                <i *ngIf="selectionDetails.categories.selection === item.id" class="material-icons md-18 float-right oppia-search-bar-category-selection-symbol">&#xE876;</i>
+              </a>
+              <a (click)="toggleSelection('categories', item.id)" *ngIf="first" (keydown)="onMenuKeypress($event, 'category', {shiftTab: ACTION_CLOSE})" class="dropdown-item">
+                <span class="protractor-test-deselected">{{ item.text | translate }}</span>
+                <i *ngIf="selectionDetails.categories.selection === item.id" class="material-icons md-18 float-right oppia-search-bar-category-selection-symbol">&#xE876;</i>
+              </a>
+              <a (click)="toggleSelection('categories', item.id)" *ngIf="last" (keydown)="onMenuKeypress($event, 'category', {tab: ACTION_CLOSE})"
+                 class="dropdown-item">
+                <span class="protractor-test-deselected">{{ item.text | translate }}</span>
+                <i *ngIf="selectionDetails.categories.selection === item.id" class="material-icons md-18 float-right oppia-search-bar-category-selection-symbol">&#xE876;</i>
+              </a>
+            </li>
+          </ng-container>
+        </ul>
+      </div>
+    </div>
+
+    <div class="oppia-same-row-container">
+      <div ngbDropdown [ngClass]="{'open' : activeMenuName === 'language', 'dropup' : enableDropup}" class="search-bar-float-left oppia-search-bar-language-selector protractor-test-search-bar-language-selector dropdown" autoClose="outside">
+        <button ngbDropdownToggle
+                (click)="openSubmenu($event, 'language')"
+                (keydown)="onMenuKeypress($event, 'language', {shiftTab: ACTION_CLOSE, enter: ACTION_OPEN})"
+                type="button"
+                class="btn protractor-test-search-bar-dropdown-toggle oppia-search-bar-language-input dropdown-toggle language-dropdown-toggle"
+                title="{{selectionDetails.languageCodes.description | translate}}">
+          <i *ngIf="isMobileViewActive()" class="material-icons md-18 language-selector-icon">&#xe8e2;</i>
+          {{ languageButtonText | truncate:14 }}
+        </button>
+        <ul ngbDropdownMenu class="oppia-search-bar-dropdown-menu-section dropdown-menu"
+            role="menu">
+          <ng-container *ngFor="let item of selectionDetails.languageCodes.masterList trackBy: index">
```

Comment:
Missing semi colon here? Just an fyi, trackBy in angular expects a function that returns a value.

Reply:
Yeah missing semicolon. OK, thanks!

Reply:
For the trackby, read through this: https://angular.io/api/common/NgForOf#description. It would look something  like this: `<li *ngFor="let item of items; index as i; trackBy: trackByFn">...</li>` Where trackByFn is a function. If you want to keep trackby with just index, I would recommend creating a util function and use it. `const trackByFn = (index: number): number => index;`

Reply:
Have raised issue https://github.com/oppia/oppia/issues/15961 for changing it across codebase.